app/lib/identity_tijuana/member_sync.rb

```diff
@@ -78,6 +147,88 @@ def self.import_user(user_id, sync_id)
       calling_subscription: ['associated_audits', 'Member_subscription', %w[unsubscribed_at]],
     }.freeze
 
+    # Find the primary TJ user match for a given ID member.
+    #
+    # Looks at the user for each given id, orders them by created at
+    # date, then selects:
+    #
+    #  1. The user with a normalised (but not cleaned), matching email, if any
+    #  2. The first subscribed user, if any
+    #  3. The first remaining user
+    def self.find_primary_user_for_member(member)
```

Comment:
I think there a number of issues here.
First, this `fix` points out that we need to have a concept of `primary` (link?) for Id's external members, which is established in the very early stages of the process, rather than being continually best-guessed.

Furthermore, the syncing process logic cannot be the same both ways.
Id's member record is the source of truth, but as the changes to multiple TJ users that point to it are possible and valid, they should be synced to the Id's member record regardless of not being a `primary` link. Eg. a linked TJ user signs the petition with typo'ed email providing an updated postcode, the new postcode is an important change and should flow into the Id's record, rather than just linking the typo'ed user.

On the other hand, and this regards also the concept of `primary`, I don't fully understand why we are not propagating the Id's record state (being a source of truth) to all linked TJ users (email attr being the exception).

Finally, the `primary` link, in the current state, should actually be the record with the cleaned email address. There reasons are:
* this is the only record that a member can personally can change via web portal
* this is the record that is used when cutting lists / sending mailouts
* this link should explicitly be marked as `primary` (connecting to the initial thought)
   * the reason for this is that if the email address is changed on Id's member record, then the email address on the `primary` should also be updated

Just for reference here – currently we have `5695` with 2, `92` with 3, and `2` with 4 TJ users.

Reply:
> I don't fully understand why we are not propagating the Id's record state (being a source of truth) to all linked TJ users (email attr being the exception).

Because I want to deploy this before I go on leave.